pkg/gzip/gzip_test.go

```diff
@@ -25,12 +28,35 @@ func TestFileWithoutDir(t *testing.T) {
 	err = os.Mkdir(dest, 0755)
 	Assert(t).IsNil(err, "expected no error creating destdir")
 
+	err = setup(dest)
+	Assert(t).IsNil(err, "expected no error performing test-specific setup")
+
 	user, err := user.Current()
 	Assert(t).IsNil(err, "expected no error getting current user")
 
 	err = ExtractTarGz(user.Username, file, dest)
-	Assert(t).IsNil(err, "expected no error extracting tarball")
 
-	_, err = os.Stat(filepath.Join(dest, "a", "b"))
-	Assert(t).IsNil(err, "expected no error statting extracted file")
+	check(err, dest)
+}
+
+func testExtraction(t *testing.T, tarfile string, check func(error, string)) {
+	testExtractionWithSetup(t, tarfile, func(string) error { return nil }, check)
```

Comment:
You may notice nothing calls `testExtractionWithSetup` with an actual setup function... I was going to use that in another test for upcoming functionality I haven't finished yet. I guess I can remove it from this PR